lib/msf/core/post/file.rb

```diff
@@ -174,9 +174,21 @@ def executable?(path)
   # @return [Boolean] true if +path+ exists and is writable
   #
   def writable?(path)
-    raise "`writable?' method does not support Windows systems" if session.platform == 'windows'
-
-    cmd_exec("test -w '#{path}' && echo true").to_s.include? 'true'
+    # for windows we write a file to ensure everything is in order
+    if session.platform == 'windows'
+      f="#{path}\\#{Rex::Text.rand_text_alphanumeric(4..8)}.txt"
+      words = Rex::Text.rand_text_alphanumeric(9)
+      begin
+        # path needs to have double, not single quotes
+        c= %Q(cmd.exe /C echo '#{words}' >> "#{f}" && type "#{f}" && del "#{f}")
```

Comment:
I really don't like the idea of implementing this in the Mixin where it would cause all modules that use it to effectively drop an IOC to disk. I think it would be fine for your module, because between the notes and the description you've made it clear that things will be written to disk. Would you be opposed to just moving this into the module until we can find a way to run the check without writing to disk? @gwillcox-r7 and I did some research and this looks non-trivial.

Reply:
So there's a time you'd want to check if a folder is writable but then not actually write something to it? I half figured if you're checking if a folder can be written to, 100% chance you're going to write to it anyways

Reply:
@h00die To give some more context to this, technically there is a way to check if you can write to a folder without writing to it using tools like AccessChk from SysInternals, which would avoid dropping IOCs on the target. However as far as we could tell, there doesn't seem to be any way to perform these checks using the built in tools that Windows offers.

Therefore our concern was more that we would be altering modules that may import this library and have different IOC requirements, so making this change could have more wide spread impact than initially anticipated. As @smcintyre-r7 mentioned we are happy to take a look at this in the future though once we have cleared up some additional details.

If you need any further clarification feel free to ask; open to opinions on this :)